boxoffice/models/invoice.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from decimal import Decimal
+from boxoffice.models import db, BaseMixin, UuidMixin
+from sqlalchemy import event
+from sqlalchemy import sql, func
+from sqlalchemy.ext.orderinglist import ordering_list
+from boxoffice.models.user import get_financial_year
+
+__all__ = ['Invoice']
+
+
+def get_latest_invoice_no(organization, jurisdiction, invoice_dt):
+    """
+    Returns the last invoice number used, 0 if no order has been invoiced yet
+    """
+    query = db.session.query(sql.functions.max(Invoice.invoice_no))\
+        .filter(Invoice.organization == organization)
+    fy_start_at, fy_end_at = get_financial_year(jurisdiction, invoice_dt)
```

Comment:
FY stands for fiscal year, not financial year. The function needs to be renamed. (Although the term financial year is fairly common too.)